main.go

```diff
@@ -163,6 +165,15 @@ func newClient(cfg *externaldns.Config) (*kubernetes.Clientset, error) {
 		return nil, err
 	}
 
+	config.WrapTransport = func(rt http.RoundTripper) http.RoundTripper {
```

Comment:
the question is if this should be an option? We can extend initialiser to support `WithMetrics` optional parameter ?